com.dynamo.cr/com.dynamo.cr.bob.test/src/com/dynamo/bob/pipeline/ShaderProgramBuilderTest.java

```diff
@@ -145,6 +145,209 @@ private void doTest(boolean expectSpirv) throws Exception {
+            assertEquals("u_lights", binding_type_member.getName());
+
+            ShaderDesc.ResourceTypeInfo lights_binding_type = shader.getTypes(binding_type_member.getType().getTypeIndex());
+            assertEquals("Light", lights_binding_type.getName());
+            assertEquals(3, lights_binding_type.getMembersCount());
+
+            ShaderDesc.ResourceMember light_member_type  = lights_binding_type.getMembers(0);
+            ShaderDesc.ResourceMember light_member_pos   = lights_binding_type.getMembers(1);
+            ShaderDesc.ResourceMember light_member_color = lights_binding_type.getMembers(2);
+
+            assertEquals("type", light_member_type.getName());
+            assertEquals(ShaderDesc.ShaderDataType.SHADER_TYPE_INT, light_member_type.getType().getShaderType());
+            assertEquals("position", light_member_pos.getName());
+            assertEquals(ShaderDesc.ShaderDataType.SHADER_TYPE_VEC3, light_member_pos.getType().getShaderType());
+            assertEquals("color", light_member_color.getName());
+            assertEquals(ShaderDesc.ShaderDataType.SHADER_TYPE_VEC4, light_member_color.getType().getShaderType());
+        }
+
+        {
+            String fs_src =
+                "#version 430 \n" +
+                "struct Data \n" +
+                "{ \n" +
+                "    vec4 member1; \n" +
+                "}; \n" +
+                "buffer Test \n" +
+                "{ \n" +
+                "    Data my_data_one; \n" +
+                "    Data my_data_two[]; \n" +
+                "}; \n" +
+                "out vec4 color_out; \n" +
+                "void main() \n" +
+                "{ \n" +
+                "    color_out = my_data_one.member1 + my_data_two[0].member1; \n" +
+                "} \n";
+
+            List<Message> outputs = build("/reflection_1.fp", fs_src);
+            ShaderDesc shaderDesc = (ShaderDesc) outputs.get(0);
+
+            assert(shaderDesc.getShadersCount() > 0);
+            ShaderDesc.Shader shader = getShaderByLanguage(shaderDesc, ShaderDesc.Language.LANGUAGE_SPIRV);
+            assertNotNull(shader);
+
+            debugPrintShader("Reflection Test 1", shader);
+
+            assert(shader.getStorageBuffersCount() == 1);
+            ShaderDesc.ResourceBinding binding_test = shader.getStorageBuffers(0);
+            assertEquals("Test", binding_test.getName());
+
+            ShaderDesc.ResourceTypeInfo binding_type = shader.getTypes(binding_test.getType().getTypeIndex());
+            assertEquals("Test", binding_type.getName());
+        }
+
+        {
+            String fs_src =
+                "#version 430\n" +
+                "uniform sampler2D sampler_2d;\n" +
+                "uniform sampler2DArray sampler_2d_array;\n" +
+                "uniform samplerCube sampler_cube;\n" +
+                // TODO:
+                //"uniform samplerBuffer sampler_buffer;\n" +
```

Comment:
We don't need to support this right now, so it's just left as a note for the future